thrift/src/main/java/com/linecorp/armeria/common/thrift/text/StackWalkingCurrentThriftMessageClassFinder.java

```diff
@@ -0,0 +1,102 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+// =================================================================================================
+// Copyright 2011 Twitter, Inc.
+// -------------------------------------------------------------------------------------------------
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this work except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file, or at:
+//
+//  https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =================================================================================================
+package com.linecorp.armeria.common.thrift.text;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Stream;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class StackWalkingCurrentThriftMessageClassFinder extends AbstractCurrentThriftMessageClassFinder {
+    private final Logger logger =
+            LoggerFactory.getLogger(StackWalkingCurrentThriftMessageClassFinder.class);
+    private final DefaultCurrentThriftMessageClassFinder defaultCurrentThriftClassFinder =
+            new DefaultCurrentThriftMessageClassFinder();
+
+    private final MethodHandle walkMH;
+    private final MethodHandle getInstanceMH;
+    private MethodHandle getClassNameMH;
+
+    StackWalkingCurrentThriftMessageClassFinder() throws Throwable {
+        final MethodHandles.Lookup lookup = MethodHandles.lookup();
+        final MethodType walkMT = MethodType.methodType(Object.class, Function.class);
+        final Class<?> stackWalker = Class.forName("java.lang.StackWalker");
+        walkMH = lookup.findVirtual(stackWalker, "walk", walkMT);
+
+        final MethodType getInstanceMT = MethodType.methodType(stackWalker);
+        getInstanceMH = lookup.findStatic(stackWalker, "getInstance", getInstanceMT);
+
+        final Class<?> stackFrame = Class.forName("java.lang.StackWalker$StackFrame");
```

Comment:
Ditto - please specify the class loader for consistent behavior, because otherwise the context class loader is used.